src/Monolog/Handler/AzureHandler.php

```diff
@@ -0,0 +1,121 @@
+r<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog\Handler;
+
+use Monolog\Level;
+use Monolog\Utils;
+use Monolog\LogRecord;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Handler\Curl;
+
+/**
+ * AzureHandler uses cURL to send log to Azure Monitor
+ * https://learn.microsoft.com/en-us/azure/azure-monitor/
+ * 
+ * @author Michele D'Amico <michele.damico@linfaservice.it>
+ * Linfa Service - https://www.linfaservice.it
+ * Damikael - https://www.damikael.dev
+ */
+class AzureHandler extends AbstractProcessingHandler
+{
+    private string $tenantId;
+    private string $appId;
+    private string $appSecret;
+    private string $dceURI;
+    private string $dcrImmutableId;
+    private string $table;
+
+    /**
+     * @param string $tenantId
+     * @param string $appId
+     * @param string $appSecret
+     * @param string $dceURI
+     * @param string $dcrImmutableId;
+     * @param string $table
+     *
+     * @throws MissingExtensionException If the curl extension is missing
+     */
+    public function __construct(string $tenantId, string $appId, string $appSecret, string $dceURI, string $dcrImmutableId, string $table, int|string|Level $level = Level::Debug, bool $bubble = true)
+    {
+        if (!\extension_loaded('curl')) {
+            throw new MissingExtensionException('The curl extension is needed to use the AzureHandler');
+        }
+
+        $this->tenantId = $tenantId;
+        $this->appId = $appId;
+        $this->appSecret = $appSecret;
+        $this->dceURI = $dceURI;
+        $this->dcrImmutableId = $dcrImmutableId;
+        $this->table = $table;
+
+        parent::__construct($level, $bubble);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function write(LogRecord $record): void
+    {
+
+        // retrieve access_token
+        $url = "https://login.microsoftonline.com/" . $this->tenantId . "/oauth2/v2.0/token";
+        $postString = "

```

Comment:
You can replace string post field by an array as follow:

```php
 $postString = [
    'grant_type' => 'client_credentials',
    'scope' => 'https://monitor.azure.com//. default',
    'client_id' => $this->appId,
    'client_secret' => $this->appSecret 
];
```

You can then remove the  `CURLOPT_HTTPHEADER` option because PHP will automatically add the header `Content-Type` to `multipart/form-data`.